**SIMULACION**

Generar 2 números pseudoaleatorios (dados), sumar estos números y generar una tabla con la suma de los mismos, su frecuencia y probabilidad. Para ello se debe generar 100, 1000 y 10000 eventos de lanzamiento de dados

In [5]:

import random 
valor = 0
lista =[ ]
for j in range(0,100):
    for i in range(0,2):
        v=random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/100)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     5            0.05
3     7            0.07
4     5            0.05
5     13            0.13
6     13            0.13
7     17            0.17
8     17            0.17
9     7            0.07
10     10            0.1
11     6            0.06


In [3]:
import random 
valor = 0
lista =[ ]
for j in range(0,1000):
    for i in range(0,2):
        v=random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/1000)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     36            0.036
3     39            0.039
4     98            0.098
5     95            0.095
6     141            0.141
7     188            0.188
8     133            0.133
9     108            0.108
10     91            0.091
11     50            0.05
12     21            0.021


In [4]:
import random 
valor = 0
lista =[ ]
for j in range(0,10000):
    for i in range(0,2):
        v=random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/10000)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     254            0.0254
3     567            0.0567
4     820            0.082
5     1135            0.1135
6     1400            0.14
7     1674            0.1674
8     1304            0.1304
9     1188            0.1188
10     838            0.0838
11     550            0.055
12     270            0.027


**NUMPY**

In [8]:
import numpy as n
valor = 0
lista =[ ]
for j in range(0,10000):
    for i in range(0,2):
        v=n.random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/10000)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     410            0.041
3     769            0.0769
4     1197            0.1197
5     1690            0.169
6     1978            0.1978
7     1579            0.1579
8     1192            0.1192
9     760            0.076
10     425            0.0425


In [7]:
import numpy as n
valor = 0
lista =[ ]
for j in range(0,1000):
    for i in range(0,2):
        v=n.random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/1000)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     44            0.044
3     100            0.1
4     117            0.117
5     198            0.198
6     180            0.18
7     156            0.156
8     104            0.104
9     51            0.051
10     50            0.05


In [6]:
import numpy as n
valor = 0
lista =[ ]
for j in range(0,100):
    for i in range(0,2):
        v=n.random.randint(1,6)
        valor=valor+v
    lista.append(valor)
    valor=0
suma=[]
repeticiones=[]
for x in set(lista):
    repeticiones.append(lista.count(x))
    suma.append(x)

    probabilidad=[]
for y in repeticiones:
    probabilidad.append(y/100)

print("suma  repeticiones  probabilidad")
for x in range(0,len(probabilidad)):
    print(str(suma[x])+"     "+str(repeticiones[x])+"            "+str(probabilidad[x]))

suma  repeticiones  probabilidad
2     6            0.06
3     5            0.05
4     14            0.14
5     11            0.11
6     22            0.22
7     18            0.18
8     16            0.16
9     3            0.03
10     5            0.05


**Conclusiones**

100 eventos: 7 y 8 repeticiones 17 probabilidad 0.17
1000 eventos: 7 repeticiones 188 probabilidad 0.18
10000 eventos: 7 repeticiones 1674 probabilidad 0.16
    
los números con más probabilidad de salir serian el 7 y 8 
